application/libraries/Api/Command/V1/SurveyResponses/conditions/DateRangeConditionHandler.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace LimeSurvey\Libraries\Api\Command\V1\SurveyResponses\conditions;
+
+use InvalidArgumentException;
+use LimeSurvey\Libraries\Api\Command\V1\SurveyResponses\HandlerInterface;
+
+class DateRangeConditionHandler implements HandlerInterface
+{
+    public function canHandle(string $operation): bool
+    {
+        if (strtolower($operation) == 'date-range') {
+            return true;
+        }
+        return false;
+    }
+
+    public function execute($key, $value): object
+    {
+        $key = preg_replace('/[^a-zA-Z0-9_-]/', '', $key);
```

Comment:
This looks like a repeated logic, it would make sense to separate this preparation of the key into a reusable method and use that in the handlers